assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java

```diff
@@ -451,9 +452,18 @@ public void handleAccess(HttpServletRequest req, HttpServletResponse res, Refere
                     // fail the request, user not logged in yet.
                 } else {
                     if (REF_TYPE_SUBMISSIONS.equals(ref.getSubType())) {
+                        String assignmentName = "bulk_download";
+                        try {
+                            assignmentName = getAssignment(ref.getReference()).getTitle();
+                        } catch (Exception ignore) {
+                            log.error("Could not find assignment for ref = {}", ref.getReference());
+                        }
+
+                        SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd_HH-mm");
+                        formatter.setTimeZone(userTimeService.getLocalTimeZone());
                         String queryString = req.getQueryString();
                         res.setContentType("application/zip");
-                        res.setHeader("Content-Disposition", "attachment; filename = bulk_download.zip");
+                        res.setHeader("Content-Disposition", "attachment; filename = \"" + assignmentName + " " + formatter.format(new Date()) + ".zip\"");
```

Comment:
I think (have seen this) that the browser handles the replacement of invalid filename characters.

Reply:
Better just to replace them rather than rely on browsers fixing them fixing them up unpredictably. (Typically with underscore replacements) I tried with an assignment with the % character and it had some problems in Edge downloading. Unless all supported browsers were verified with all "characters to avoid", I'm sure it will be a future bug.